### ISSUE TITLE
Disallow coders from directly pushing commits

### ISSUE THREAD

**Issue by ghost** [d3athrow/vgstation13#7430]
Currently there are some bold people pushing their code without peer review or feedback from community directly into the bleeding edge. This unsurprisingly results in drama and shitcode which could be easily prevented should a dialogue take place beforehand.

Exceptions would be as @PJB3005 said https://github.com/d3athrow/vgstation13/issues/7353#issuecomment-168159934

> Also, can we fucking disallow coders from directly pushing shit unless it's either fixes or OH GOD things

**Comment by 9600bauds** [comment]
ultimately they're the repository owners, so other than a soft-worded plea of consideration (which this isn't) then we can't force anything

**Comment by ghost** [comment]
Apathy is death.

**Comment by Exxion** [comment]
OK but Bauds is right
It's literally impossible to force the repo owners to not directly push commits
We can only make an agreement and hope they follow it

**Comment by ghost** [comment]
It's impossible to force people to fix bugs, we still make bug reports.

**Comment by Exxion** [comment]
Which is analogous to the "make an agreement and hope they follow it" part of my comment.

**Comment by ghost** [comment]
Termed, it's the same reason we're on bleeding edge instead of master. It works.

**Comment by AutismSpider** [comment]
Better start working on fucking fixing bugs form there shit code we have all most 250+ bug reports

**Comment by Intigracy** [comment]
Autism, what bugs are specifically caused by @N3X15 or @D3athrow (wow look I'm not pretending we're not talking about specific people)? Because usually when they cause an issue, if you let them know about it they get to it as soon as they can

This isn't even about the direct pushing, considering they could just make a PR and then instantly merge it after travis checks it. This is about an executive decision by the two project leads being made that you disagree with.

After people stopped deliberately sabotaging the maps to try getting it changed back, people calmed down and started making genuine feedback reports (like how now the maps have minimum and maximum playercounts) and bug reports (since they actually have to play the maps now, they can find specific things they find wrong with them).

I give it at most a week before it's just the minority of loudspoken whiners that are left complaining about it, where everyone else'll adapt and probably find a new map to enjoy.

tl;dr fuck off termed

DYK: Back in 2013, the UI was changed to the current setup. Player opinions weren't taken into account until after the UI change, and everything turned out for the better because the old one was much shittier compared to this one.

**Comment by AutismSpider** [comment]
not just map bugs we have shit like this #7412 #7337 and so on #7336 three issue about slimes so far gos on lets just fix the little things 1st then the maps most of the map stuff is reworking the thing or adding in missing stuff.

**Comment by ghost** [comment]
@Intigracy 
I hope you're not expecting a serious response after making a strawman of the issue and then comparing UI improvements to forcing people to play maps they don't want, while simultaneously telling me to fuck off.

**Comment by gbasood** [comment]
I just find it funny that this issue was created by someone who has never directly contributed to the codebase, and the issue itself is a complaint about code management.

**Comment by ghost** [comment]
@gbasood
What is funny is that a community derived from an anonymous posting board still discredits arguments based on the poster. You'd think such irrational behavior wouldn't be present.

**Comment by Intigracy** [comment]
> This isn't even about the direct pushing, considering they could just make a PR and then instantly merge it after travis checks it. This is about an executive decision by the two project leads being made that you disagree with.

**Comment by ihadtoregisterforthis** [comment]
It's not just the name, it's the reputation and intentions carried along that keep appearing predictably. The bitter speech, the insults and name calling, explicit and implicit. Denial of other opinions and avoiding the point, starting flame wars, criticizing others with equally criticizable quotes. The original discussion is lost and it's a battle of people instead of opinions. Symptoms of shitpost even when there is a valid point deep below. 

---

Content PRs should have some time before being added, even better if they have feedback. It's a matter of etiquette and interest of the effect on the game, and also reviewing possible code issues.
Many rushed merges got equally fast reverts, and it's something that could have been avoided.

**Comment by 9600bauds** [comment]
> What is funny is that a community derived from an anonymous posting board still discredits arguments based on the poster. You'd think such irrational behavior wouldn't be present.

Are you serious right now?
You got an actual, honest response. The "fuck off termed" is just a one line at the end of the completely unrelated response. You're trying to handwave the entire post away because ONE LINE out of five serious paragraphs memed on you.

**Comment by ghost** [comment]
@9600bauds
I did not even dignify the "fuck off termed" with a response to it.

**Comment by clusterfack** [comment]
Because you know that he is completely right

**Comment by PJB3005** [comment]
@N3X15 recently pushed AAC arithmetic functions, which were relatively pathed, something we disallowed **years** ago.

Not to mention the fucking CMC is _still_ buggy as hell.

**Comment by Arthurdentist** [comment]
as far as bug fixes and the like goes I see no problem with them being auto merged. Content PRs should never be merged immediately without due consideration. I think this whole map vote buisness was handled really poorly and at the very least it would have been a good idea to ask around in OOC and see what people there think. Even if you despise the thread you can still ask on the server and see what the people actually playing think of it. Admins can easily setup votes as well and if I or any other admin had been asked to do so then we would have gladly done it.

I've not really got sympathy for nexis. If you can't handle having the thread say mean things about you then you should either try and handle controversial changes more delicately or just fuck off to a different station.

**Comment by clusterfack** [comment]
The CMC wasn't pushed

**Comment by AutismSpider** [comment]
@Arthurdentist  oh get real the thread is useless when it comes to stuff like this better off asking ooc for stuff like this

**Comment by gbasood** [comment]
@TermedClepe

> bluh bluh anonymous posters

What a completely irrelevant argument! I'm talking about you, and I know who you are, so why does anonymity matter here? Stop trying to ignore the actual complaint I have against you opening this issue.

And why you should care about how we maintain a codebase that you have _literally nothing to do with_ is beyond me, and we definitely don't need you to tell us how to do it properly.

If you took issue with an aspect of code that was pushed without a PR I would understand. If you had opened an issue against that instead of how it was introduced into the codebase, I would understand. But this is so completely not important to someone who does not actually touch the code that it makes no sense why this is an issue for you.

**Comment by ghost** [comment]
@gbasood
An attack on an argument made by attacking the character, motive, or other attribute of the person making the argument, rather than attacking the argument directly.

**Comment by 9600bauds** [comment]
Why are you complaining about one remark about you when you are yet to address the serious, on point responses?
Either this is a disinformation tactic or your point fell apart by now.

**Comment by ghost** [comment]
@9600bauds
Please quote which point you'd like me to address.

**Comment by 9600bauds** [comment]
Everything you've handwaved away by saying "I hope you're not expecting a serious response".
Apparently this entire issue is not for serious discussion.

**Comment by ghost** [comment]
@9600bauds 
If you're having difficulties quoting points you want me to address, then you might see why I have not addressed them in the first place. But by all means continue to bicker at my persona you imagined me to have.

**Comment by gbasood** [comment]
I say we close the issue so we can continue ignoring TermedClepe and we can discuss this again if the actual 'reason' for it being opened continues to happen in a detrimental manner

**Comment by ghost** [comment]
Since the offending party actually left the project, I see no reason to keep it open myself. Unless you guys want to continue throwing fallacies at me.

**Comment by Intigracy** [comment]
Just proving my point that this was just about the one change you didn't like and not the actual OP on the issue, as shown by your complete lack of care about pomf continuing to direct push (which wasn't and still isn't a problem).

**Comment by 9600bauds** [comment]
pretty much

**Comment by clusterfack** [comment]
Wow holy shit the attempting to rabble rouse to fight a completely separate issue, what the fuck is wrong with you termed

**Comment by Exxion** [comment]
https://youtu.be/5DmYLrxR0Y8

**Comment by d3athrow** [comment]
utts butts coconuts, this is a valid thing and for any content merges i'll be making PRs from now on. bug fixes hastily written on a whim will still be 420forcepusherrday though.
